aioredis/commands/streams.py

```diff
@@ -0,0 +1,222 @@
+import sys
+from collections import OrderedDict
+
+from aioredis.util import wait_convert, wait_make_dict, wait_ok
+
+PY_VER = sys.version_info
+
+if PY_VER < (3, 0):
+    from itertools import izip as zip
+
+
+def fields_to_dict(fields):
+    """Convert a flat list of key/values into an OrderedDict"""
+    fields_iterator = iter(fields)
+    return OrderedDict(zip(fields_iterator, fields_iterator))
+
+
+def parse_messages(messages):
+    """ Parse messages as returned by Redis into something useful
+
+    Messages returned by XRANGE arrive in the form:
+
+        [
+            [message_id, [key1, value1, key2, value2, ...]],
+            ...
+        ]
+
+    Here we parse this into:
+
+        [
+            [message_id, OrderedDict(
+                (key1, value1),
+                (key2, value2),
+                ...
+            )],
+            ...
+        ]
+
+    """
+    if messages is None:
+        return []
+    return [(mid, fields_to_dict(values)) for mid, values in messages]
+
+
+def parse_messages_by_stream(messages_by_stream):
+    """ Parse messages returned by stream
+
+    Messages returned by XREAD arrive in the form:
+        [stream_name,
+            [
+                [message_id, [key1, value1, key2, value2, ...]],
+                ...
+            ],
+            ...
+        ]
+
+    Here we parse this into (with the help of the above parse_messages()
+    function):
+
+        [
+            [stream_name, message_id, OrderedDict(
+                (key1, value1),
+                (key2, value2),.
+                ...
+            )],
+            ...
+        ]
+
+    """
+    if messages_by_stream is None:
+        return []
+
+    parsed = []
+    for stream, messages in messages_by_stream:
+        for message_id, fields in parse_messages(messages):
+            parsed.append((stream, message_id, fields))
+    return parsed
+
+
+def parse_lists_to_dicts(lists):
+    """ Convert [[a, 1, b, 2], ...] into [{a:1, b: 2}, ...]"""
+    def _list_to_dict(list_):
```

Comment:
This function is almost the same as `fields_to_dict` except return type,
It is better to make a single function and pass `OrderedDict`/dict` types as argument.

Reply:
Ok, will do